### PR TITLE
Const all the things

### PR DESCRIPTION
**Pull request by elazarg**
Also some modernization. Essentially, variable, parameters and methods should be const by default.

## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced const-correctness in various functions and loops, preventing unintended modifications to variables.
- **New Features**
	- Improved error handling with clearer messages for marshaling and unmarshaling processes.
	- Added new test cases for edge cases and specific instruction behaviors in the marshaling tests.
- **Documentation**
	- Updated comments for clarity in several functions and methods.
- **Refactor**
	- Simplified parameter types in function signatures across multiple files for improved readability and performance.
- **Tests**
	- Expanded test cases for eBPF program verification and marshaling functionality.

### CODE REVIEW

File: src/crab/type_domain.cpp [hunk 1]
```diff
@@ -194,7 +194,7 @@ void TypeDomain::selectively_join_based_on_type(NumAbsDomain& dst, NumAbsDomain&
     dst |= std::move(src);
 
     // Now add in the extra invariants saved above.
-    for (auto& [variable, interval] : extra_invariants) {
+    for (const auto& [variable, interval] : extra_invariants) {
         dst.set(variable, interval);
     }
```
[the remark this answers is not in the document]
Comment (elazarg) [reply src/crab/type_domain.cpp:199]: ???
